core/state/snapshot/conversion.go

```diff
@@ -322,7 +322,7 @@ func generateTrieRoot(db ethdb.KeyValueWriter, it Iterator, account common.Hash,
 						return
 					}
 					if !bytes.Equal(account.Root, subroot.Bytes()) {
-						results <- fmt.Errorf("invalid subroot(%x), want %x, got %x", it.Hash(), account.Root, subroot)
+						results <- fmt.Errorf("invalid subroot(%x), want %x, got %x", hash, account.Root, subroot)
```

Comment:
This error message `invalid subroot(%x)` says to me that the `%x` in there is the invalid root. Gonna make things harder to understand. Perhaps: `"invalid subroot (path %x): want %x, got %x"`